pkg/operator2/status.go

```diff
@@ -33,6 +33,28 @@ func (c *authOperator) setFailingStatus(operatorConfig *operatorv1.Authenticatio
 	return err
 }
 
+func (c *authOperator) setProgressingStatus(operatorConfig *operatorv1.Authentication, reason, message string) error {
```

Comment:
I am wondering if we call this too much - i.e. if we should build up a series of conditions and apply once?

Reply:
Actually it should not matter because the helper functions only update when needed.